PWGGA/GammaConv/AliAnalysisTRDEfficiency.cxx

```diff
@@ -304,700 +148,470 @@ void AliAnalysisTRDEfficiency::UserCreateOutputObjects()
+    Double_t ppid = 0; 
+    Double_t psag = 0;
+    Double_t prat = 0; 
+    Double_t ptrd = 0;
+    Double_t ptkl = 0;
+    Double_t npt = 0; 
+    Double_t npid = 0;
+    Double_t nsag = 0;
+    Double_t nrat = 0;
+    Double_t ntrd = 0;
+    Double_t ntkl = 0;
+    Double_t gpt  = photon->GetPhotonPt();
+    Double_t gR   = photon->GetConversionRadius();
+    Double_t geta = photon->GetPhotonEta();
+    Double_t gphi = photon->GetPhotonPhi();        
+
+    for (Int_t j = 0; j < fESD->GetNumberOfTrdTracks(); j++){
+        AliESDTrdTrack *trdtrack = fESD->GetTrdTrack(j);
+        if (!trdtrack) continue;
+        if (trdtrack->GetTrackMatch() == (AliVTrack*)ptrack){   // positive track match
+            Int_t fLayerMask = trdtrack->GetLayerMask();
+            Double_t rating = GetRating(V0, ptrack, trdtrack);
+            ptrd = (fLayerMask >> 0) & 1;       // does this give a boolean to track being in the 0th layer
+            ppt  = trdtrack->Pt();
+            ppid = trdtrack->GetPID();
+            psag = GetSagitta(trdtrack);
+            prat = rating;
+            ptkl = trdtrack->GetNTracklets();
+        }
+        if (trdtrack->GetTrackMatch() == (AliVTrack*)ntrack){   // negative track match
+            Int_t fLayerMask = trdtrack->GetLayerMask();
+            Double_t rating = GetRating(V0, ntrack, trdtrack);
+            ntrd = (fLayerMask >> 0) & 1;
+            npt  = trdtrack->Pt();
+            npid = trdtrack->GetPID();
+            nsag = GetSagitta(trdtrack);
+            nrat = rating;
+            ntkl = trdtrack->GetNTracklets();
+        }
+    }
+    tmp[0] = ppt;
+    tmp[1] = ppid;
+    tmp[2] = psag;
+    tmp[3] = prat;
+    tmp[4] = ptrd;
+    tmp[5] = ptkl;
+    
+    tmp[6] = npt;
+    tmp[7] = npid;
+    tmp[8] = nsag;
+    tmp[9] = nrat;
+    tmp[10]= ntrd;
+    tmp[11]= ntkl;
+                         
+    tmp[12]= gpt;
+    tmp[13]= gR;
+    tmp[14]= geta;
+    tmp[15]= gphi;
+                         
+    return tmp;
+}
+
+Int_t AliAnalysisTRDEfficiency::GetEventCuts(AliESDTrdTrack *trdtrack, TString clss){
+    
+    if (trdtrack->Pt() < 2)         // pt cut
+        return 1;  // only trd track
+    
+    if (TMath::Abs(GetSagitta(trdtrack))>0.2) // sag cut
+        return 2; // pass pt
+    
+    if (trdtrack->GetPID() < 164)   // pid cut
+        return 3; // pass sag
+
+    if (trdtrack->GetNTracklets() < 5)// # of tracklet cuts
+        return 4; // pass pid
 
+    Int_t fLayerMask = trdtrack->GetLayerMask(); // layer 0 cuts
+    if (( (fLayerMask >> 0) & 1) == 0 )
+        return 5; // 
+    
+    if (!clss.Contains("CINT7-T"))
+        return 6;
+    
+    if (!clss.Contains("HQU"))
+        return 7;
+    
+    return 8;
+}
 
+//_____________________________________________________________________________
+Bool_t AliAnalysisTRDEfficiency::GetTrackCuts(AliESDtrack *track){
+    //
+    static AliESDtrackCuts *EsdTrackCuts = 0x0;
+    EsdTrackCuts = new AliESDtrackCuts();

```

Comment:
I do not see that GetTrackCuts is used, but if you call this method it may give you a memory leak - this new never gets deleted, which is ok if you check the existence of the object before calling this function

Reply:
It seems to be a relic from a previous build. I dont call it anymore.

Reply:
ok, for the next commit you may consider to remove it - but for now I'll approve this PR